File: app/src/main/java/org/pragyan/dalal18/fragment/marketDepth/MarketDepthFragment.java
package org.pragyan.dalal18.fragment.marketDepth;

import android.graphics.Color;
import android.os.Bundle;
import androidx.annotation.NonNull;
import com.google.android.material.tabs.TabLayout;
import androidx.fragment.app.Fragment;
import androidx.core.content.ContextCompat;
import androidx.viewpager.widget.ViewPager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import org.pragyan.dalal18.R;
import org.pragyan.dalal18.adapter.ContentAdapter;

public class MarketDepthFragment extends Fragment {

    public MarketDepthFragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        View customView =  inflater.inflate(R.layout.fragment_market_depth, container, false);

        // Binding view for Tabbed Fragments
        ViewPager viewPager = customView.findViewById(R.id.content_viewPager);
        TabLayout tabLayout = customView.findViewById(R.id.tab_headings);

        // Set adapter to viewpager and custom colors to tabLayout
        viewPager.setAdapter(new ContentAdapter(getChildFragmentManager()));
        tabLayout.setupWithViewPager(viewPager);

        if (getContext() != null) {
            tabLayout.setTabTextColors(
                    ContextCompat.getColor(getContext(), R.color.neutral_font_color),
                    ContextCompat.getColor(getContext(), R.color.neon_blue));
        }
        tabLayout.setBackgroundColor(Color.parseColor("#20202C"));

        return customView;
    }
}

File: app/src/main/java/org/pragyan/dalal18/fragment/marketDepth/MarketDepthFragment.kt
package org.pragyan.dalal18.fragment.marketDepth

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentStatePagerAdapter
import org.pragyan.dalal18.R
import org.pragyan.dalal18.adapter.pagerAdapters.DepthPagerAdapter
import org.pragyan.dalal18.databinding.FragmentMarketDepthBinding
import org.pragyan.dalal18.utils.viewLifecycle

class MarketDepthFragment : Fragment() {

    private var binding by viewLifecycle<FragmentMarketDepthBinding>()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        binding = FragmentMarketDepthBinding.inflate(inflater, container, false)

        binding.apply {
            // Set Adapter to Viewpager
            contentViewPager.adapter = DepthPagerAdapter(childFragmentManager, FragmentStatePagerAdapter.BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT)
            tabHeadings.setupWithViewPager(contentViewPager)
        }

        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        (activity as AppCompatActivity).supportActionBar?.title = resources.getString(R.string.market_depth)
    }
}
